### PR TITLE
Fix object composition

### PR DESCRIPTION
**Pull request by kunjsong01**
Fixed 991 issue 2 as in https://github.com/esbmc/esbmc/issues/991#issuecomment-1525504379

Turn
```cpp
this->itr = iterator();  // no argument for iterator constructor
```
into: 
```cpp
iterator(&this->itr);
```

Also removed unnecessary recursive type getters if a class/struct type already exists in symbol table.

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]

File: src/clang-c-frontend/clang_c_convert.cpp [hunk 1]
```diff
@@ -3417,3 +3416,16 @@ bool clang_c_convertert::get_vft_binding_expr(
   abort();
   return true;
 }
+
+void clang_c_convertert::get_ref_to_struct_type(typet &type)
+{
+  /*
+   * For some special cases, we need to get a symbol type referring to
```
Comment (rafaelsamenezes) [inline src/clang-c-frontend/clang_c_convert.cpp:3423]: Do you have any example?

File: src/clang-cpp-frontend/clang_cpp_convert.h [hunk 2]
```diff
@@ -171,6 +183,18 @@ class clang_cpp_convertert : public clang_c_convertert
...
+   * A default copy constructor would have the return type below:
+   * * return_type: constructor
+   *   #default_copy_cons: 1
+   *
+   * Arguments:
+   *  cxxmdd: clang AST node representing the ctor/dtor we are dealing with
+   *  rtn_type: the corresponding return type node
+   */
+  void
+  annotate_cdtor_rtn_type(const clang::CXXMethodDecl *cxxmdd, typet &rtn_type);
```
Comment (rafaelsamenezes) [inline src/clang-cpp-frontend/clang_cpp_convert.h]: This needs a better name.

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]
Just some nitpicks

File: src/clang-c-frontend/clang_c_convert.cpp (hunk 1 above)
Comment (kunjsong01) [reply src/clang-c-frontend/clang_c_convert.cpp:3423]: Fixed - added an example in the comment.

File: src/clang-cpp-frontend/clang_cpp_convert.h (hunk 2 above)
Comment (kunjsong01) [reply src/clang-cpp-frontend/clang_cpp_convert.h]: Fixed - disambiguate and replaced the name with "annotate_ctor_dtor_rtn_type"

**Review by rafaelsamenezes — APPROVED** [review]

**Pull request merged**